lxd/storage/drivers/driver_zfs.go

```diff
@@ -365,14 +365,15 @@ func (d *zfs) Validate(config map[string]string) error {
 
 			return validate.IsBool(value)
 		}),
-		"zfs.export":                  validate.Optional(validate.IsBool),
-		"volume.zfs.blocksize":        validate.Optional(ValidateZfsBlocksize),
-		"volume.zfs.remove_snapshots": validate.Optional(validate.IsBool),
-		"volume.zfs.use_refquota":     validate.Optional(validate.IsBool),
-		"volume.zfs.reserve_space":    validate.Optional(validate.IsBool),
+		"zfs.export": validate.Optional(validate.IsBool),
 	}
 
-	return d.validatePool(config, rules)
+	volRules := d.volumeRules()
+
+	// Override 'zfs.blocksize' validator as this one is dependant from volume on volume level.
+	volRules["zfs.blocksize"] = validate.Optional(ValidateZfsBlocksize)
```

Comment:
I don't understand why this bit is needed? The comment doesn't make sense to me.
It feels like if we need these sorts of exceptions then the model for solving this isn't quite right or we are missing something.

Reply:
This is a special case where `volume.zfs.block.size` for pool and `zfs.blocksize` for volume have different validators. `validate.Optional(ValidateZfsBlocksize)` and `validate.Optional(ValidateZfsVolBlocksize(vol))` respectively, so we can't use volume validator on pool level because we don't have access to volume instance. But I think that we can do this in old way, this will be clearer.